front_end/content.py

```diff
@@ -776,27 +794,145 @@ def get_help_request(self, course_id, assignment_id, exercise_id, user_id):
+                          ON r.user_id = u.user_id
+                        INNER JOIN exercises e
+                          ON e.course_id = r.course_id
+                          AND e.assignment_id = r.assignment_id
+                          AND e.exercise_id = r.exercise_id
+                        WHERE r.course_id = ?
+                          AND NOT r.user_id = ?
+                          AND e.output_type = ?'''
+                requests = self.fetchall(sql, (course_id, user_id, row["output_type"]))
+                sim_dict = []
+
+                #jpg output uses the diff_jpg function in helper.py, txt output uses .similarity() from the Spacy module
+                if row["output_type"] == "img":
+                    for request in requests:
+                        diff_image, diff_percent = diff_jpg(orig_output, request["image_output"])
+                        if diff_percent < .10:
+                            request_info = {"psim": 1 - diff_percent, "course_id": request["course_id"], "assignment_id": request["assignment_id"], "exercise_id": request["exercise_id"], "user_id": request["user_id"], "name": request["name"], "student_comment": request["student_comment"],  "code": request["code"], "text_output": request["text_output"], "suggestion": request["suggestion"]}
+                            sim_dict.append(request_info)
+
+                else:
+                    nlp = spacy.load('en_core_web_sm')
+                    orig = nlp(orig_output)
+
+                    for request in requests:
+                        curr = nlp(request["text_output"])
+                        psim = curr.similarity(orig)
+                        sim = False
+
+                        #these thresholds can be changed in the future
+                        if len(orig) < 10 and len(curr) < 10:
+                            if psim > .30:
+                                sim = True
+                        elif len(orig) < 100 and len(curr) < 100:
+                            if psim > .50:
+                                sim = True
+                        elif len(orig) < 200 and len(curr) < 200:
+                            if psim > .70:
+                                sim = True
+                        else:
+                            if psim > .90:
+                                sim = True
+
+                        if sim:
+                            request_info = {"psim": psim, "course_id": request["course_id"], "assignment_id": request["assignment_id"], "exercise_id": request["exercise_id"], "user_id": request["user_id"], "name": request["name"], "student_comment": request["student_comment"],  "code": request["code"], "text_output": request["text_output"], "suggestion": request["suggestion"]}
+                            sim_dict.append(request_info)
+                            
+                #self.test_similarity()
+                return sim_dict
+
+    def test_similarity(self):
```

Comment:
I kept it just in case we wanted to test some more and see what would be the "sweet spot" for similarity percentages, but I'll go ahead and remove it because we can easily update that as we go.